File: WebApplication1/WebApplication1/DTO/Response/ConcertResDto.cs
using System.Text.Json.Serialization;

namespace WebApplication1.DTO.Response
{
    public class ConcertResDto
    {
        public int ConcertId { get; set; }
        public required string ConcertName { get; set; }
        public  ConcertSpecDto ConcertSpec { get; set; } 
        public int CreatorId { get; set; }
     
        public string? HallName { get; set; }
       
        public string? CustomHall { get; set; }
        public ICollection<HallBookingDto> HallBookings { get; set; } = null!;    
    }
}

File: angular-app/src/app/api/models/concert-res-dto.ts
/* tslint:disable */
/* eslint-disable */
import { ConcertSpecDto } from './concert-spec-dto';
import { HallBookingDto } from './hall-booking-dto';
export interface ConcertResDto {
  concertId?: number;
  concertName: null | string;
  concertSpec?: ConcertSpecDto;
  creatorId?: number;
  customHall?: null | string;
  hallBookings?: null | Array<HallBookingDto>;
  hallName?: null | string;
}
